lib/medtronic/processData.js

```diff
@@ -638,19 +615,170 @@ var buildBasalRecords = function (records) {
+         the battery is left out for too long. The strategy is to filter out any
+         time change to a date where the year is less than the current year
+         minus one. This is the same strategy used in the CareLink driver. */
+      if (fromDatum.jsDate.getUTCFullYear() < (new Date().getUTCFullYear() - 1)) {
+        console.log('Excluding time change to', toDatum.jsDate.toISOString().slice(0,-5), 'as spurious.');
+        continue;
+      }
+
+      var timeChange = cfg.builder.makeDeviceEventTimeChange()
+        .with_change({
+          from: sundial.formatDeviceTime(fromDatum.jsDate),
+          to: sundial.formatDeviceTime(toDatum.jsDate),
+          agent: 'manual'
+        })
+        .with_deviceTime(sundial.formatDeviceTime(toDatum.jsDate))
+        .set('jsDate', toDatum.jsDate)
+        .set('index', toDatum.index);
+      postrecords.push(timeChange);
+    } else {
+      console.log('Records not matching:',fromDatum, toDatum);
+      throw new Error('Missing record');
+    }
+
+  };
+  cfg.tzoUtil = new TZOUtil(cfg.timezone, mostRecent, postrecords);
+  return { postrecords : postrecords, tzoUtil : cfg.tzoUtil };
+}
+
+function buildAlarmRecords(records) {
+
+  var alarmRecords = filterHistory([RECORD_TYPES.ALARM_PUMP, RECORD_TYPES.LOW_RESERVOIR, RECORD_TYPES.LOW_BATTERY], records);
+  var postrecords = [];
+
+  var getAlarmName = function (idx,types) {
+    return common.getName(types, idx);
+  };
+
+  alarmRecords.forEach(function (alarmdatum) {
+
+    console.log('alarm:',alarmdatum.jsDate, common.bytes2hex(alarmdatum.head), common.bytes2hex(alarmdatum.date), common.bytes2hex(alarmdatum.body));
+
+    var alarmRecord = cfg.builder.makeDeviceEventAlarm()
+      .with_deviceTime(sundial.formatDeviceTime(alarmdatum.jsDate))
+      .set('index', alarmdatum.index);
+    cfg.tzoUtil.fillInUTCInfo(alarmRecord, alarmdatum.jsDate);
+
+    switch(alarmdatum.type.value) {
+      case RECORD_TYPES.ALARM_PUMP.value:
+        var alarmValue = struct.extractByte(alarmdatum.head,1);
+        var alarmText = getAlarmName(alarmValue,ALARM_TYPES);
+
+        switch (alarmValue) {
+          case ALARM_TYPES.ALARM_NO_DELIVERY.value:
+            alarmRecord = alarmRecord.with_alarmType('no_delivery');
+            break;
+          case ALARM_TYPES.ALARM_BATTERY_DEPLETED.value:
+            alarmRecord = alarmRecord.with_alarmType('no_power');
+            break;
+            case ALARM_TYPES.ALARM_AUTO_OFF.value:
+              alarmRecord = alarmRecord.with_alarmType('auto_off');
+              break;
+          default:
+            alarmRecord = alarmRecord.with_alarmType('other');
+            var payload = {alarm_id: alarmValue};
+            if(alarmText !== 'unknown') {
+              payload.alarm_text = alarmText;
+            }
+            alarmRecord = alarmRecord.with_payload(payload);
+        }
+        break;
+      case RECORD_TYPES.LOW_RESERVOIR.value:
```

Comment:
I just checked the CSV and indeed the amount is listed there, so I'll try to identify it in the raw data and add to payload.